dhcpv4/nclient4/client.go

```diff
@@ -129,39 +166,58 @@ type Client struct {
 
 // New returns a client usable with an unconfigured interface.
 func New(iface string, opts ...ClientOpt) (*Client, error) {
-	i, err := net.InterfaceByName(iface)
-	if err != nil {
-		return nil, err
-	}
-	pc, err := NewRawUDPConn(iface, ClientPort)
-	if err != nil {
-		return nil, err
-	}
-	return NewWithConn(pc, i.HardwareAddr, opts...)
+	return new(iface, nil, nil, opts...)
```

Comment:
Just out of curiosity, is there a reason for reorganizing this?

Reply:
Yes. The `Conn` should be initialized after processing options to be able to initialize an unicast `Conn` via these options.

Reply:
Thanks for the clarification!